helper/cert/source_gen.go

```diff
@@ -138,15 +139,14 @@ func (s *GenSource) Certificate(ctx context.Context, last *Bundle) (Bundle, erro
 
 	// Generate cert, set it on the result, and return
 	cert, key, err := s.generateCert()
-	if err == nil {
-		result.Cert = []byte(cert)
-		result.Key = []byte(key)
+	if err != nil {
+		return result, err
```

Comment:
I think this is probably technically ok because the result is discarded if there's an error, but I'll let Theron confirm.

Reply:
It is indeed being discarded in `notify.Run()` if there's an error. Which is good, because the next block there is comparing the new result to the old one, so it would also think an empty result is a new result. All this to say I think we might as well leave it up to the caller when returning an error, and technically `s.caCert` is valid at this point and will continue to be used on the next pass.